### ISSUE TITLE
AudioElement.loop doesn't work on Firefox 50+

### ISSUE THREAD

**Issue by bbaudel-gpsw** [GoogleChrome/omnitone#32]
If the sound has a loop property, the sound file is playing but there is no sound output.

```
var audioElement = document.createElement('audio');
audioElement.loop = true;
```

This issue is also verified when calling `audioElement.pause();` when the end of sound is reached then 
```
audioElement.currentTime = 0;
audioElement.play();
```

Can be verified on this sample : http://labs.plan8.se/ambisonics-webplayer/ with 0.1.8 version.

**Comment by hoch** [comment]
Not sure if this is Omnitone's issue though. Omnitone does not affect any aspect of AudioElement. It simply uses the stream from the element. Can you repro the issue in FF without Omnitone? If so, I think you have file a bug to the FireFox bug tracker.

**Comment by hoch** [comment]
@bbaudel-gpsw Any feedback?

**Comment by bbaudel-gpsw** [comment]
@hoch I've no sound issue with a simple AudioElement which has the loop attribute or which is played when audioElement is ended.
But when Omnitone is activated and the sound is looped I've no sound anymore into Firefox browser but the audio file continue (currentTime is changing) without any sound. Works on Chrome.
So seems to be an Omnitone implementation issue or an AudioContext loss.

**Comment by hoch** [comment]
Okay, let me break down this issue:

> I've no sound issue with a simple AudioElement which has the loop attribute or which is played when audioElement is ended.

Yes, this is how it should be.

> the audio file continue (currentTime is changing)

You can't tell if the audio file continues or not. The context currentTime should be definitely running, but that does not mean that you're getting a valid stream from the audio/video element. One thing you can do is to plug an analyser node directly after MediaElementSourceNode in Omnitone. However, I doubt that you would get anything from it.

> So seems to be an Omnitone implementation issue or an AudioContext loss.

I am still not convinced and I have to build a repro case to confirm it. What do you mean by "AudioContext loss"?

See here:
https://github.com/GoogleChrome/omnitone/blob/gh-pages/src/js/demo-player.js#L530

The demo player uses .play(), .pause(), and .currentTime on the video element. It plays the audio correctly in FF (v50).

**Comment by hoch** [comment]
I just updated FF to v51 and the Omnitone demo went silent right after I pressed 'pause'. So I can confirm this is a regression from FireFox because it works fine on v50.

You can find FF v50.1 here: https://ftp.mozilla.org/pub/firefox/releases/50.1.0/

**Comment by hoch** [comment]
@bbaudel-gpsw Thanks for catching this. It's really hard to cover all the browsers since I am the only one who actively works on this project.

**Comment by bbaudel-gpsw** [comment]
Good analysis. By the way, it's a very nice job on Omnitone library. It's always difficult to act for all browser engines changes.

This FF issue can be the problem ?
https://bugzilla.mozilla.org/show_bug.cgi?id=1188496

**Comment by padenot** [comment]
> @padenot Paul, should I file an issue to bugzilla on this? Any recent change in the media element?

Yeah I just got aware of this yesterday, for some reason 51 (and only 51, 52 and up work fine) breaks `HTMLMediaElement` + `MediaElementAudioSourceNode` + setting `currentTime` on the `HTMLMediaElement`. I'm currently looking into the cause and figuring out what to do.

This is tracked in https://bugzilla.mozilla.org/show_bug.cgi?id=1337653

**Comment by hoch** [comment]
@padenot Thanks so much, Paul!

I'll leave this issue open to track the progress.

**Comment by hoch** [comment]
@bbaudel-gpsw According the issue on bugzilla, this seems to be fixed. Can you verify in your application?

**Comment by bbaudel-gpsw** [comment]
Fixed in FF 52.0.1. Thanks to the Mozilla team.
